### ISSUE TITLE
Compile errors not appearing for rust

### ISSUE THREAD

**Issue by jknoetzke** [helix-editor/helix#5405]
### Summary

<img width="811" alt="image" src="https://user-images.githubusercontent.com/97972/210645288-f09b1791-2708-4e25-ad06-88ff120dd4b9.png">

### Reproduction Steps

Install rust-analyzer
Open a rust file in Helix.

### Helix log

```
2023-01-04T21:01:17.358 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":1,"result":null}
2023-01-04T21:01:17.358 helix_lsp::transport [INFO] <- null
2023-01-04T21:01:17.933 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"range":{"end":{"character":0,"line":0},"start":{"character":0,"line":0}},"text":"/"}],"textDocument":{"uri":"file:///Users/jfk/Projects/aws-lambda-rust-runtime/examples/http-cors/src/main.rs","version":1}}}
2023-01-04T21:01:17.946 helix_vcs::diff::worker [INFO] Diff computation timed out, update of diffs might appear delayed
2023-01-04T21:01:18.076 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"range":{"end":{"character":1,"line":0},"start":{"character":1,"line":0}},"text":"/"}],"textDocument":{"uri":"file:///Users/jfk/Projects/aws-lambda-rust-runtime/examples/http-cors/src/main.rs","version":2}}}
2023-01-04T21:01:18.089 helix_vcs::diff::worker [INFO] Diff computation timed out, update of diffs might appear delayed
2023-01-04T21:01:42.137 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"range":{"end":{"character":2,"line":0},"start":{"character":2,"line":0}},"text":":"}],"textDocument":{"uri":"file:///Users/jfk/Projects/aws-lambda-rust-runtime/examples/http-cors/src/main.rs","version":3}}}
2023-01-04T21:01:42.149 helix_vcs::diff::worker [INFO] Diff computation timed out, update of diffs might appear delayed
2023-01-04T21:01:42.153 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/completion","params":{"position":{"character":3,"line":0},"textDocument":{"uri":"file:///Users/jfk/Projects/aws-lambda-rust-runtime/examples/http-cors/src/main.rs"}},"id":2}
2023-01-04T21:01:42.156 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":2,"result":null}
2023-01-04T21:01:42.157 helix_lsp::transport [INFO] <- null
2023-01-04T21:01:42.374 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"range":{"end":{"character":3,"line":0},"start":{"character":3,"line":0}},"text":"q"}],"textDocument":{"uri":"file:///Users/jfk/Projects/aws-lambda-rust-runtime/examples/http-cors/src/main.rs","version":4}}}
2023-01-04T21:01:42.386 helix_vcs::diff::worker [INFO] Diff computation timed out, update of diffs might appear delayed
2023-01-04T21:01:42.809 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"range":{"end":{"character":4,"line":0},"start":{"character":4,"line":0}},"text":"!"}],"textDocument":{"uri":"file:///Users/jfk/Projects/aws-lambda-rust-runtime/examples/http-cors/src/main.rs","version":5}}}
2023-01-04T21:01:46.498 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"shutdown","params":null,"id":3}
2023-01-04T21:01:46.499 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":3,"result":null}
2023-01-04T21:01:46.501 helix_lsp::transport [INFO] <- null
2023-01-04T21:01:46.501 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"exit","params":null}
```

### Platform

macOS

### Terminal Emulator

iTerm Build 3.4.19

### Helix Version

helix 22.12 (96ff64a8)

**Comment by the-mikedavis** [comment]
It looks like this is not the full log. Could you submit the beginning part where Helix sends the `initialize` message?

Also, is this a public project? Having a reproduction project would help with debugging.

**Comment by jknoetzke** [comment]
Clone: https://github.com/awslabs/aws-lambda-rust-runtime

Head into any of the example directories `aws-lambda-rust-runtime/examples/http-query-parameters` (for instance) and open up a rust file.

Now, what I have noticed is that if you open up a rust file in `aws-lambda-rust-runtime/lambda-http` it works fine.

I attached the full logs.



[helix.log](https://github.com/helix-editor/helix/files/10361192/helix.log)
